testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -3839,6 +3839,18 @@ public void testPredicateOnRowTypeField()
         }
     }
 
+    @Test
+    public void testPredicateOnTwoRowsWithTheSameFieldNames()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE) && hasBehavior(SUPPORTS_INSERT) && hasBehavior(SUPPORTS_ROW_TYPE));
+
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_predicate_on_two_rows", "(row1 row(x INT), row2 row(x INT))")) {
```

Comment:
Please make sure this test fails without #15408 in Iceberg connector using Parquet format. I'm sure it passes even without the commit.

Reply:
I think the table structure should be:

```suggestion
        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_predicate_on_two_rows", "(x INT, row1 row(x INT))")) {
```

Reply:
@bluzy I think case you are proposing is covered in testPredicateOnRowTypeField. Isn't it?

Reply:
How about this query into `testPredicateOnRowTypeField` ?

`assertQuery("SELECT int_t FROM " + table.getName() + " WHERE int_t = 2 AND row_t.int_t = 1", "VALUES 1");`